packages/client/src/components/CreatePost.js

```diff
@@ -1,96 +1,117 @@
 import { useState } from "react";
-import ButtonDiv from "./ButtonDiv";
-import Button from "./Button";
-import InputDiv from "./InputDiv";
-import Input from "./Input";
-import InputLabel from "./InputLabel";
-import TextArea from "./TextArea";
-import Checkbox from "./Checkbox";
-import Form from "./Form";
-import FormHeader from "./FormHeader";
+import { Redirect } from "react-router-dom";
+import ButtonDiv from "./styled/ButtonDiv";
+import Button from "./styled/Button";
+import InputDiv from "./styled/InputDiv";
+import Input from "./styled/Input";
+import InputLabel from "./styled/InputLabel";
+import TextArea from "./styled/TextArea";
+import Form from "./styled/Form";
+import FormHeader from "./styled/FormHeader";
+import Container from "./styled/Container";
+import TextLink from "./styled/TextLink";
+
+// Post Model looks like:
+// title: {
+//   type: String,
+//   required: true,
+// },
+// body: {
+//   type: String,
+//   required: true,
+// },
+// isDraft: {
+//   type: Boolean,
+//   required: true,
+// },
+// publishedAt: {
+//   type: Date,
+// },
 
-// Data looks like this (for backend)
-// console.log({
-//     title:'string',
-//     body:'string',
-//     isDraft:boolean
-// })
 const backend = process.env.REACT_APP_BACKEND;
-const endpoint = `${backend}/api/createPost`;
+const endpoint = `${backend}/api/create-post`;
 function CreatePost() {
-  const [title, setTitle] = useState("");
-  const [body, setBody] = useState("");
-  const [isDraft, setIsDraft] = useState(true);
+  const [post, setPost] = useState({
+    title: "",
+    body: "",
+    isDraft: true,
+  });
+
+  const [redirect, setRedirect] = useState(false);
+
+  const createPostObject = (e) => {
+    const { name, value } = e.target;
+    setPost({ ...post, [name]: value });
+  };
 
   const handleSubmit = async (e) => {
     // we might want to go to the next page to see the published post (have to verify this)
     e.preventDefault();
+
+    setPost({ ...post, isDraft: e.target.value });
+    console.log(post);
+
     // Question: Do we need form validation here for Leon?
-    try {
-      const res = await fetch(endpoint, {
-        method: "POST",
-        headers: {
-          "Content-type": "application/json",
-        },
-        body: JSON.stringify({
-          title,
-          body,
-          isDraft,
-        }),
-      });
-    } catch (error) {
-      console.log(error);
-    }
+    // try {
+    //   const res = await fetch(endpoint, {
+    //     method: "POST",
+    //     headers: {
+    //       "Content-type": "application/json",
+    //     },
+    //     body: JSON.stringify({ post }),
+    //   });
+    // } catch (error) {
+    //   console.log(error);
+    // }
   };
-
+  if (redirect) {
+    return <Redirect to="/" />;
+  }
```

Comment:
This is probably not the way to do this, might just be better to pull the history object off of `useHistory` and push a new url into it.